contracts/modules/commons/submodules/auth/SequenceBaseSig.sol

```diff
@@ -0,0 +1,185 @@
+
+          // Write weight and address to image
+          bytes32 node = _joinAddrAndWeight(addr, addrWeight);
+          root = root != bytes32(0) ? LibOptim.fkeccak256(root, node) : node;
+          continue;
+        }
+
+        if (flag == FLAG_SIGNATURE) {
+          // Read weight
+          uint8 addrWeight;
+          (addrWeight, rindex) = _signature.readUint8(rindex);
+
+          // Read single signature and recover signer
+          uint256 nrindex = rindex + 66;
+          address addr = SignatureValidator.recoverSigner(_subdigest, _signature[rindex:nrindex]);
+          rindex = nrindex;
+
+          // Acumulate total weight of the signature
+          weight += addrWeight;
+
+          // Write weight and address to image
+          bytes32 node = _joinAddrAndWeight(addr, addrWeight);
+          root = root != bytes32(0) ? LibOptim.fkeccak256(root, node) : node;
+          continue;
+        }
+
+        if (flag == FLAG_DYNAMIC_SIGNATURE) {
+          // Read signer and wight
+          uint8 addrWeight; address addr;
+          (addrWeight, addr, rindex) = _signature.readUint8Address(rindex);
+
+          // Read signature size
+          uint256 size;
+          (size, rindex) = _signature.readUint24(rindex);
+
+          // Read dynamic size signature
+          uint256 nrindex = rindex + size;
+          if (!SignatureValidator.isValidSignature(_subdigest, addr, _signature[rindex:nrindex])) {
+            revert InvalidNestedSignature(_subdigest, addr, _signature[rindex:nrindex]);
+          }
+          rindex = nrindex;
+
+          // Acumulate total weight of the signature
+          weight += addrWeight;
+
+          // Write weight and address to image
+          bytes32 node = _joinAddrAndWeight(addr, addrWeight);
+          root = root != bytes32(0) ? LibOptim.fkeccak256(root, node) : node;
+          continue;
+        }
+
+        if (flag == FLAG_NODE) {
+          // Read node hash
+          bytes32 node;
+          (node, rindex) = _signature.readBytes32(rindex);
+          root = root != bytes32(0) ? LibOptim.fkeccak256(root, node) : node;
+          continue;
+        }
+
+        if (flag == FLAG_BRANCH) {
+          // Enter a branch of the signature merkle tree
+          uint256 size;
+          (size, rindex) = _signature.readUint24(rindex);
+          uint256 nrindex = rindex + size;
+
+          uint256 nweight; bytes32 node;
+          (nweight, node) = recoverBranch(_subdigest, _signature[rindex:nrindex]);
+
+          weight += nweight;
+          root = LibOptim.fkeccak256(root, node);
+
+          rindex = nrindex;
+          continue;
+        }
+
+        if (flag == FLAG_SUBDIGEST) {
+          // A hardcoded always accepted digest
+          // it pushes the weight to 100%
+          bytes32 hardcoded;
+          (hardcoded, rindex) = _signature.readBytes32(rindex);
+          if (hardcoded == _subdigest) {
+            weight = type(uint256).max;
```

Comment:
Small question: Is this going to be OK if the weight can overflow? My guess is it's safe because worst case scenario is that a valid signature is identified as invalid, but want to confirm if this is what you think too.

Reply:
I think it's okay.